### ISSUE TITLE
IDE0301 reported for expression code

### ISSUE THREAD

**Issue by MaceWindu** [dotnet/roslyn#71121]
**Version Used**: 4.8.0-7.23572.1+7b75981cf3bd520b86ec4ed00ec156c8bc48e4eb

**Steps to Reproduce**:

```cs
using System.Linq.Expressions;

public class Test
{
	public byte[] Property { get; set; }
	public Expression<Func<Test>> Method() => () => new Test() { Property = Array.Empty<byte>() };
}
```

**Diagnostic Id**: IDE0301

**Expected Behavior**: no diagnostics regarding new C# features from expressions if they are not supported there

**Actual Behavior**:

proposed fix results in compilation error
```cs
using System.Linq.Expressions;

public class Test
{
	public byte[] Property { get; set; }

        // CS9175 An expression tree may not contain a collection expression
	public Expression<Func<Test>> Method() => () => new Test() { Property = [] };
}
```

**Comment by CyrusNajmabadi** [comment]
I believe this was also just fixed.

**Comment by MaceWindu** [comment]
probably #71012, didn't found initially as it doesn't mention IDE0301

**Comment by CyrusNajmabadi** [comment]
Dupe of https://github.com/dotnet/roslyn/issues/71012
